### PR TITLE
Misc

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java [hunk 1]
```diff
@@ -124,13 +124,27 @@ protected BaseOperation(OkHttpClient client, Config config, String apiGroup, Str
...
+   * Separates apiVersion for apiGroup/apiVersion combination.
+   * @param apiVersion  The apiVersion or apiGroup/apiVersion combo.
+   * @return            Just the apiVersion part without the apiGroup.
+     */
+  private static String trimVersion(String apiVersion) {
+    if (apiVersion == null) {
+      return null;
+    } else {
+      String[] versionParts = apiVersion.split("/");
+      return versionParts.length > 1 ? versionParts[versionParts.length - 1] : apiVersion;
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java]: You don't need to do that check, just always return `versionParts[versionParts.length - 1]` - that's what I was trying to say before! See https://repl.it/DfTr/0 for proof.

File: openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java [hunk 2]
```diff
@@ -1,12 +1,12 @@
 /**
  * Copyright (C) 2015 Red Hat, Inc.
- *
+ * <p>
```
Comment (jimmidyson) [inline openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java]: Dodgy chars due to IDEA formatting.

File: openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java [hunk 3]
```diff
@@ -34,5 +42,77 @@ public DeploymentConfigOperationsImpl(OkHttpClient client, OpenShiftConfig confi
...
+
+    public DeploymentConfigReaper(DeploymentConfigOperationsImpl operation, OkHttpClient client) {
+      this.operation = operation;
+      this.client = client;
+    }
+
+    @Override
+    public void reap() {
+      DeploymentConfig deployment = operation.cascading(false).edit().editSpec().withReplicas(0).endSpec().done();
+      waitForObservedGeneration(deployment.getStatus().getObservedGeneration());
```
Comment (jimmidyson) [inline openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java]: If we release kube-model & upgrade we should be able to work with available replicas - see https://github.com/fabric8io/kubernetes-model/blob/master/kubernetes-model/src/main/resources/schema/kube-schema.json#L8639-L8673

**Comment by iocanel** [comment]
[merge]

File: openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java (hunk 3 above)
Comment (jimmidyson) [inline openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java]: You seem to have ignored this?

**Pull request merged**